infrastructure/w3id/src/logs/log-manager.ts

```
@@ -0,0 +1,142 @@
+		let verified = false;
+		if (!proof) throw new Error();
+		for (const key of currentUpdateKeys) {
+			const signValidates = await verifyCallback(
+				canonicalJson as string,
+				proof,
+				key,
+			);
+			if (signValidates) verified = true;
+		}
+		if (!verified) throw new BadSignatureError();
+	}
+
+	private async appendEntry(entries: LogEvent[], options: RotationLogOptions) {
+		const { signer, nextKeyHashes, nextKeySigner } = options;
+		const latestEntry = entries[entries.length - 1];
+		const logHash = await hash(latestEntry);
+		const index = Number(latestEntry.versionId.split("-")[0]) + 1;
+
+		const currKeyHash = await hash(nextKeySigner.pubKey);
+		if (!latestEntry.nextKeyHashes.includes(currKeyHash))
+			throw new BadNextKeySpecifiedError();
+
+		const logEvent: LogEvent = {
+			id: latestEntry.id,
+			versionTime: new Date(Date.now()),
+			versionId: `${index}-${logHash}`,
+			updateKeys: [nextKeySigner.pubKey],
+			nextKeyHashes: nextKeyHashes,
+			method: "w3id:v0.0.0",
+		};
+
+		const proof = await signer.sign(canonicalize(logEvent) as string);
+		logEvent.proof = proof;
+
+		await this.repository.create(logEvent);
+		return logEvent;
+	}
+
+	private async createGenesisEntry(options: GenesisLogOptions) {
+		const { id, nextKeyHashes, signer } = options;
+		const logEvent: LogEvent = {
+			id,
+			versionId: `0-${id.split("@")[1]}`,
+			versionTime: new Date(Date.now()),
+			updateKeys: [signer.pubKey],
+			nextKeyHashes: nextKeyHashes,
+			method: "w3id:v0.0.0",
+		};
+		const proof = await signer.sign(canonicalize(logEvent) as string);
+		logEvent.proof = proof;
+		await this.repository.create(logEvent);
+		return logEvent;
+	}
+
+	async createLogEvent(options: CreateLogEventOptions) {
+		const entries = await this.repository.findMany({});
+		if (entries.length > 0)
+			return this.appendEntry(entries, options as RotationLogOptions);
+		return this.createGenesisEntry(options as GenesisLogOptions);
+	}
```

if entries length is 0, and i happen to pass rotation log options instead of genesis log options, this should throw an error about it receiving unexpected options? 

i mean ofc, that would be a stupid thing to do by the consumer of this lib, but as a lib, i think that error might be good to have